rfcs/rdm-0109-python-versions.md

```diff
@@ -0,0 +1,168 @@
+| 2025-10         | post Python 3.14 release — pre RDMv14 release — development       | 3.14                         | 3.14                          |
+| 2026-08         | RDMv14 release — users + development                              | 3.14                         | 3.14                          |
+| 2026-10         | post Python 3.15 release — pre RDMv15 release — development       | 3.14                         | 3.15                          |
+| 2026-10         | post Python 3.15 release — pre RDMv15 release — users             | 3.14                         | 3.14                          |
+| 2027-08         | RDMv15 release — users + development                              | 3.14                         | 3.15                          |
+| ...             | ...                                                               | 3.14                         | ...                           |
+| 2028-10 (e.g.,) | post Python 3.17 release, decided to be new minimum — development | 3.17                         | 3.17                          |
+| ...             | ...                                                               | 3.17                         | ...                           |
+| 2030-10         | post Python 3.19 release — pre RDMv19 release — development       | 3.19                         | 3.19                          |
+
+Although a version of InvenioRDM could un-officially support a newly released version of Python, official support is only guaranteed
+when a dedicated release is made that officially supports it.
+
+#### Continuous integration (CI) tests
+The CI should test the minimum version required and the maximum supported. Skip in-between versions as they would have been tested in the past and this keeps the CI efficient in aggregate *and* per individual PR.
+
+#### Reasoning
+By adopting this policy, we ...
+
+- ... benefit from the longest support window by jumping to 3.14 as minimum version for RDMv14
+- ... take advantage of Python performance gains as a baseline
+- ... don't have to change minimum version every year and constantly churn
+- ... are alerted and account for upcoming deprecations in development as soon as official next latest Python version is out
+- ... align Python and InvenioRDM version numbers which is nice
+
+### InvenioRDM — periods of overlap with deprecated Python versions
+
+The Python support policy outlined above leads to a scenario where a newly released InvenioRDM and the previous, but still supported for 6 months per [RDM support policy](https://inveniordm.docs.cern.ch/releases/maintenance-policy/), do not share the same mimimum required Python version for that overlap period. This could be a rare occurrence (e.g., every 5 years), but it has to be accounted for.
+
+As is, it would mean that during that period of time, any fix would have to be compatible with the previous minimum Python version — a different and unsupported (by Python community) one.
```

Comment:
This is taking the "first option" mentioned in previous version of this RFC: 

Maintainers continue for 6 months to make fixes and backports that are compatible with the previous minimum Python version. This means being more careful about fixes and it doesn't promote adoption of a secure/supported version of Python. It is simpler because it doesn't require anything else. Past the 6 months mark, all the new features of the new minimum Python version can be adopted (and all deprecation workarounds that can be dropped are).